zebra-chain/src/orchard/tree.rs

```diff
@@ -228,6 +261,49 @@ impl NoteCommitmentTree {
     pub fn uncommitted() -> pallas::Base {
         pallas::Base::one().double()
     }
+
+    /// Count of note commitments added to the tree.
+    ///
+    /// For Orchard, the tree is capped at 2^32.
+    pub fn count(&self) -> u32 {
+        self.parents.iter().enumerate().fold(
+            match (self.left, self.right) {
+                (None, None) => 0,
+                (Some(_), None) => 1,
+                (Some(_), Some(_)) => 2,
+                (None, Some(_)) => unreachable!(),
+            },
+            |acc, (i, p)| {
+                // Treat occupation of parents array as a binary number
+                // (right-shifted by 1)
+                acc + if p.is_some() { 1 << (i + 1) } else { 0 }
+            },
+        )
+    }
+
+    fn is_complete(&self) -> bool {
+        self.left.is_some()
+            && self.right.is_some()
+            && self.parents.len() == MERKLE_DEPTH - 1
+            && self.parents.iter().all(|p| p.is_some())
+    }
+}
+
+impl From<Vec<pallas::Base>> for NoteCommitmentTree {
+    /// Compute the tree from a whole bunch of note commitments at once.
+    fn from(values: Vec<pallas::Base>) -> Self {
+        let mut tree = Self::default();
+
+        if values.is_empty() {
+            return tree;
+        }
+
+        for cm_x in values {
+            let _ = tree.append(cm_x);
+        }
+
+        tree
+    }
 }
```

Comment:
Code style: Would this be clearer as an `Extend` impl? Do we want this conversion and an `Extend` impl?

We should try to match whatever we've done with the history trees.
@conradoplg what do you think?

Reply:
HistoryTree does not use `Extend` because we need to return errors there. But it can be done here. I'll work on it

Reply:
Ah, in our case it can return an error too (tree full). In `From` that is not a problem because it creates a new tree from a vector, so it's unlikely to be called with 2^32 vector. But with `Extend` we'd need to handle that.

Reply:
I don't mind what we do here, happy to leave this choice until later, unless we need to extend now?